main/pipeline/BUILD

```diff
@@ -30,6 +30,7 @@ cc_library(
         "//local_vars",
         "//main/autogen",
         "//main/options",
+        "//main/pipeline/definition_checker",
```

Comment:
"definition checker" is a pretty vague name. For example, it sounds a lot like a folder we already have, `definition_validator/`.

Do we need this to be in its own folder at all? Could the file be simply `main/pipeline/DefinitionLinesDenylistEnforcer.{h,cc}` and then we include that file directly wherever it needs to be, with no bazel changes and with no new folders?